src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTests.cs

```diff
@@ -583,14 +628,25 @@ private static void VerifySimpleDecrypt(byte[] encodedMessage, CertLoader certLo
                 if (cert == null)
                     return; // Sorry - CertLoader is not configured to load certs with private keys - we've tested as much as we can.
 
-                X509Certificate2Collection extraStore = new X509Certificate2Collection(cert);
-                ecms.Decrypt(extraStore);
+#if netcoreapp // API not present on netfx
```

Comment:
If literally every test is being duplicated, it might make things cleaner to

* Change this test class to abstract, change all static Fact/Theory methods to instance.
* Make an abstract Decrypt method with whatever signature is needed
* Make two derived types, one for the existing way, and a second for explicit private key.
* Put the second class in `#if netcoreapp` (or a conditionally included .cs file).

That's how some of the Array vs Span tests were done in other libraries.

Reply:
@bartonjs, didn't know XUnit works correctly with attributes on abstract classes. Will update

Reply:
Seems like there is some issue with static methods on abstract class having Fact/Theory but it's easy to workaround